aplib.net-demo/Assets/Scripts/WFC/Grid.cs

```diff
@@ -76,5 +112,118 @@ public void PlaceRoom(int x, int y, Room room)
             this[x, y].Tile = room;
             this[x, y].Candidates = new List<Tile>();
         }
+
+        /// <summary>
+        /// Given a cell, this method returns all directly adjacent cells, excluding diagonal neighbours.
+        /// When the cell is on an edge, it will return less than 4 neighbours.
+        /// </summary>
+        /// <param name="cell">The centre cell, who's neighbours need to be determined.</param>
+        /// <returns>All neighbouring cells, excluding diagonal neighbours.</returns>
+        /// <remarks>This does not check if the tiles in the cells are connected.</remarks>
+        /// <seealso cref="GetConnectedNeighbours"/>
+        public ICollection<Cell> Get4NeighbouringCells(Cell cell)
+        {
+            ICollection<Cell> neighbours = new Collection<Cell>();
+            if (cell.X > 0)          neighbours.Add(this[cell.X - 1, cell.Y]);
```

Comment:
Even though it looks nice, please refrain from using this type of indenting. It doesn't really hurt readability if you just use one space, and makes it easier for the next programmer on what to do when they need to add a big if statement.